libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/internal/extensions/ContextEx.kt

```diff
@@ -0,0 +1,43 @@
+@file:JvmName("ContextEx")
+
+package com.mapbox.navigation.dropin.internal.extensions
+
+import android.app.Activity
+import android.content.Context
+import android.content.ContextWrapper
+import androidx.activity.ComponentActivity
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.ViewModelStoreOwner
+import com.mapbox.navigation.utils.internal.logW
+import java.lang.ref.WeakReference
+
+tailrec fun Context.recursiveUnwrap(): Context =
+    if (this !is Activity && this is ContextWrapper) {
+        this.baseContext.recursiveUnwrap()
+    } else {
+        this
+    }
+
+fun Context.toLifecycleOwner(): LifecycleOwner {
```

Comment:
It would be good to have these internal

Context is far beyond mapbox navigation domain so downstream devs will have access

Reply:
Yeah made them internal previously, but we have resorted to use `internal/extensions` as package constructs if you look at `libnavigation-base`. So used the same package construct for consistency
@kmadsen